src/utils.ts

```diff
@@ -1147,13 +1147,64 @@ export function isRecord(
   value: unknown,
   requiredKeys: string[] | undefined = undefined
 ): value is Record<string, any> {
-  const isRecord = !!value && typeof value === 'object' && !Array.isArray(value);
-  if (isRecord && requiredKeys) {
+  const isObject = (v: any) => Object.prototype.toString.call(v) === '[object Object]';
+  const hasOwnProperty = Object.prototype.hasOwnProperty;
+  if (!isObject(value)) {
+    return false;
+  }
+
+  const ctor = (value as any).constructor;
+  if (
+    ctor &&
+    (!isObject(ctor.prototype) ||
+      (ctor.prototype && !hasOwnProperty.call(ctor.prototype, 'isPrototypeOf')))
+  ) {
```

Comment:
This condition is worthy of a comment, took me a min to take it in. What's the goal of checking for the existence of isPrototypeOf on the constructor's prototype?